### ISSUE TITLE
Camera notification doesn`t work when using additional sensors (macro and wide-angle)

### ISSUE THREAD

**Issue by CapralLex** [FunkyMuse/Vigilante#64]
Xiaomi Mi 10 Lite, Android 10, MIUI Global 12.0.3 Stable

For example:
<img src="https://user-images.githubusercontent.com/38138509/103781058-dff15500-5046-11eb-9b6f-6a3cc9f1fdb4.jpg" width=35% height=35%>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<img src="https://user-images.githubusercontent.com/38138509/103781105-eed80780-5046-11eb-84a9-91ca32c5a8bb.jpg" width=35% height=35%>

**Comment by FunkyMuse** [comment]
Oh boy, here goes Android's broken implementation again

**Comment by FunkyMuse** [comment]
We'd be needing more confirmation from other devices than Xiaomi, might be Xiaomi issue only

**Comment by FunkyMuse** [comment]
LG G5, working fine on wide and normal camera

I'll try LG V20 once it charges a bit to power up but I'd doubt anything will change since it's the same camera setup

**Comment by FunkyMuse** [comment]
Yup, LG V20 and LG G5 don't suffer from the same issue, we'd need more people to give us results.

**Comment by CapralLex** [comment]
Tested on Galaxy A50 (Android 10, One UI 2.5) - works great with all sensors.
Tested on Xiaomi Mi Note 3 (Android 9, MIUI 12.0.1) - works only with front camera. Doesn't work with two rear sensors (lol)

Seems like the problem is only with Xiaomi devices  D:

Btw, microphone notifications on all devices work great

**Comment by AnonymousSeal** [comment]
OnePlus does only show, that the camera is being used, when it is in video mode. When camera or portrait mode is selected it doesn't show, that the camera is being used.

**Comment by FunkyMuse** [comment]
> Tested on Galaxy A50 (Android 10, One UI 2.5) - works great with all sensors.
> Tested on Xiaomi Mi Note 3 (Android 9, MIUI 12.0.1) - works only with front camera. Doesn't work with two rear sensors (lol)
> 
> Seems like the problem is only with Xiaomi devices D:
> 
> Btw, microphone notifications on all devices work great



> OnePlus 5 does only show, that the camera is being used, when it is in video mode. When camera or portrait mode is selected it doesn't show, that the camera is being used.

These problems seems like they're tired to OEMs not on the developer side.

**Comment by FunkyMuse** [comment]
I think i might have found a solution but apparently I don't have such device to test it, is someone willing to do so?

**Comment by FunkyMuse** [comment]
Moved to cameraX detector, hopefully it works with this issue, this is gonna be launched into the new version 1.1.0 till then the issue will be reopened, if someone can clone the code and test it, it'd be great.

**Comment by Mr-Spandan** [comment]
Yeah this is a real issue , while using macro as well as wide  as It doesn't detect the camera

**Comment by FunkyMuse** [comment]
I'll release a new update later today or tomorrow, hoping the aforementioned commit has the issue fixed.

**Comment by FunkyMuse** [comment]
> Yeah this is a real issue , while using macro as well as wide as It doesn't detect the camera

Can you check 1.1.0 and see if the issue is still there?

**Comment by Mr-Spandan** [comment]
Yeah the bug is still there.

**Comment by FunkyMuse** [comment]
After digging in the source code for CameraManager availability I think this is a bug on Google's side, I might open an issue soon if I have some way to reproduce it on a real device, it sucks that I don't have a device that contains the same problem, maybe someone cal help?

**Comment by rKsanu2MMYvypWePtQWM** [comment]
Pocophone F1, using the 2nd rear cam (used for depth information on portraits, but can be accessed with 3rd party apps) does not show the dot. Using the infrared camera in the front does not show the dot.

**Comment by geekypanda411** [comment]
Poco M2 Pro, Using Depth camera Sensor also does not notify regarding camera use.

**Comment by pokegamer5547** [comment]
I'm using Redmi note 9 pro, still having the issue on the latest update.

**Comment by FunkyMuse** [comment]
> I'm using Redmi note 9 pro, still having the issue on the latest update.

Seems like some OEMs badly implemented the camera API :(

**Comment by FunkyMuse** [comment]
Or google's camera manager still lacks detection for multiple camera sensors

**Comment by pokegamer5547** [comment]
> Or google's camera manager still lacks detection for multiple camera sensors

I feel like it's the Google camera manager doens't detect the multiple sensors. But I maybe wrong tho I'm not an developer. But I know it's annoying and hard to fix on developer side without having real device :(

**Comment by FunkyMuse** [comment]
> > Or google's camera manager still lacks detection for multiple camera sensors
> 
> I feel like it's the Google camera manager doens't detect the multiple sensors. But I maybe wrong tho I'm not an developer. But I know it's annoying and hard to fix on developer side without having real device :(

I looked deeper into the issue, it seems that google's camera manager is to blame, either the implementation done by Google or by the OEM you bought the phone from, the best way to address this is to either open an issue on the google issue tracker or contact the OEMs that have these issues, sadly I can't go furthermore into details since I don't own such device.

If someone with those type of devices that have this issue knows about Android development or is able to try and help, I can guide them through the process.

I'm labeling the issue with help needed, in case someone does want to help

**Comment by pokegamer5547** [comment]
> > > Or google's camera manager still lacks detection for multiple camera sensors
> > 
> > 
> > I feel like it's the Google camera manager doens't detect the multiple sensors. But I maybe wrong tho I'm not an developer. But I know it's annoying and hard to fix on developer side without having real device :(
> 
> I looked deeper into the issue, it seems that google's camera manager is to blame, either the implementation done by Google or by the OEM you bought the phone from, the best way to address this is to either open an issue on the google issue tracker or contact the OEMs that have these issues, sadly I can't go furthermore into details since I don't own such device.
> 
> If someone with those type of devices that have this issue knows about Android development or is able to try and help, I can guide them through the process.
> 
> I'm labeling the issue with help needed, in case someone does want to help

Oh but I'm not sure how can I issue to Google issue tracker and give them all the technical details like this. And miui (my phone skin) is pretty bad in taking feedback,I once tried to report some bugs but i never got any response or fix even after year. I would like to help you but sadly I don't know much about these things :( I hope somebody will help fixing this issue.

**Comment by gabsoftware** [comment]
Same issue with OnePlus 9 Pro. It does not detect camera usage for Panorama, Slow motion, Professional and Selfie modes.

**Comment by rajsharma91** [comment]
> I'm using Redmi note 9 pro, still having the issue on the latest update.

Redmi Note 9 Pro (Indian Variant)
Android 11 RKQ1.200826.002
MIUI 12.0.4.0 (RJWINXM) Stable
Security Patch: 01-03-2021

In my case...
1. The indicator is not visible in the rear camera of portrait mode.
2. Macro and wide-angle mode is also not working.

**Comment by FunkyMuse** [comment]
Tested on Google Pixel 5 and it works, I wish I had other devices to test onto....

**Comment by FunkyMuse** [comment]
After getting my hands on a Redmi Note 9 Pro, it seems that these bastards can't properly implement the [Camera Manager ](https://developer.android.com/reference/android/hardware/camera2/CameraManager) that Google provides, smh.

**Comment by pokegamer5547** [comment]
> After getting my hands on a Redmi Note 9 Pro, it seems that these bastards can't properly implement the [Camera Manager ](https://developer.android.com/reference/android/hardware/camera2/CameraManager) that Google provides, smh.

Damn :( those idiots, i can't believe why they change or don't properly implement the core android functionality but i noticed it too a lot of times! Like the android 10 notiffication player for youtube to seek the video directly didn't worked at all until miui 12 for me. And also now after recent update i'm no longer able yo change the priority level for notifications making those annoying icons always stay on status bar and a lot of other things. I'm never gonna buy a xaiomi phone again after this phone software updates are gone, i hate when native android features which are meant to be in every phone gets removed, yet a normal user who wouldn't even notice it because no one knows until they use it.

**Comment by FunkyMuse** [comment]
> > After getting my hands on a Redmi Note 9 Pro, it seems that these bastards can't properly implement the [Camera Manager ](https://developer.android.com/reference/android/hardware/camera2/CameraManager) that Google provides, smh.
> 
> Damn :( those idiots, i can't believe why they change or don't properly implement the core android functionality but i noticed it too a lot of times! Like the android 10 notiffication player for youtube to seek the video directly didn't worked at all until miui 12 for me. And also now after recent update i'm no longer able yo change the priority level for notifications making those annoying icons always stay on status bar and a lot of other things. I'm never gonna buy a xaiomi phone again after this phone software updates are gone, i hate when native android features which are meant to be in every phone gets removed, yet a normal user who wouldn't even notice it because no one knows until they use.it. ik programming is hard but breaking the stuff which are meant to be aready working on vanilla android (which every OEM uses as base) is ridiculous.

I guess Google should enforce their rules a bit more so that we can get the same functionalities across all devices, we'll see if Google's implementation of this application work when these devices get Android 12 update, since Google implemented a similar feature when mic/camera is used directly into the OS.

**Comment by Newcool1230** [comment]
Just a heads up looks like this issue is also occurring on the Samsung S20+.

Only the "Photo" button won't show the dot. Everything else (Single Take, video, pro, selfie cam, portrait, slow motion, etc.) is all working.

**Comment by FunkyMuse** [comment]
Update: Pixel 6 PRO, every mode works, it seems to be OEM related as expected.
